app/src/main/java/com/codingblocks/cbonlineapp/adapters/LeaderboardListAdapter.kt

```diff
@@ -1,54 +1,35 @@
 package com.codingblocks.cbonlineapp.adapters
 
-import android.content.Context
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.ArrayAdapter
-import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
 import com.codingblocks.cbonlineapp.R
 import com.codingblocks.onlineapi.models.Leaderboard
-import org.jetbrains.anko.AnkoLogger
-import java.util.*
+import kotlinx.android.synthetic.main.leader_board_list.view.*
 
-class LeaderboardListAdapter(internal var context: Context,
-                             var list: ArrayList<Leaderboard>?
-) : ArrayAdapter<Leaderboard>(context, 0), AnkoLogger {
+class LeaderboardListAdapter(val dataset: ArrayList<Leaderboard>) :
+    RecyclerView.Adapter<LeaderboardListAdapter.LeaderboardViewHolder>() {
```

Comment:
See implementation of notification recycler view follow that one for future PR's